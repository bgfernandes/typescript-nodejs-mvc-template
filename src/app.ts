import express from 'express';
import config from './config/config';

const app = express();
const port = config.port;

app.get('/', (_req : express.Request, res: express.Response) => {
  res.send('Hello World! Running on ' + config.env);
});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
app.use((error: any, _req: express.Request, res: express.Response, next: express.NextFunction) => {
  console.error(error);

  if (res.headersSent) {
    return next(error);
  }

  res.status(500).send('INTERNAL SERVER ERROR');
});

app.listen(port, () => {
  return console.log(`Server is listening on http://localhost:${port}`);
});
