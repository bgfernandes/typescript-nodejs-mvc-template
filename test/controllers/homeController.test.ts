import request from 'supertest';
import app from './../../src/app';

describe('index', () => {
  it('renders the index page', async () => {
    const response = await request(app).get('/');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  it('renders the index page localized', async () => {
    const response = await request(app).get('/?lang=pt-br');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
});
