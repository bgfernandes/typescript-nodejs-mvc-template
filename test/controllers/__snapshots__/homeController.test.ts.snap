// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders the index page 1`] = `
"
<!DOCTYPE html>
<html>
  <head>
    <title>Typescript Node MVC Template</title>

    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link href=\\"/tailwind.css\\" rel=\\"stylesheet\\">
  </head>

  <body>
    <header class=\\"flex flex-row justify-between border-b p-3\\">
  Typescript Node MVC Template
  <div class=\\"relative\\">
    <input type=\\"checkbox\\" id=\\"language-selector\\" class=\\"hidden absolute\\">
    <label for=\\"language-selector\\" class=\\"flex items-center space-x-1 cursor-pointer\\">
      <span>en</span>
      <svg class=\\"h-4 w-4\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\" />
      </svg>
    </label>

    <div id=\\"languages-menu\\" class=\\"absolute mt-1 right-1 top-full min-w-max shadow rounded opacity-0 border border-gray-400 transition delay-75 ease-in-out z-10 bg-white\\">
      <ul class=\\"block text-right\\">
        
          <li><a href=\\"?lang=en\\" class=\\"block px-3 py-2\\">en</a></li>
        
          <li><a href=\\"?lang=pt-br\\" class=\\"block px-3 py-2\\">pt-br</a></li>
        
      </ul>
    </div>
  </div>
</header>


    <div class=\\"p-3\\">
     
Hello World! Running on test

    </div>
  </body>
</html>
"
`;

exports[`index renders the index page localized 1`] = `
"
<!DOCTYPE html>
<html>
  <head>
    <title>Template MVC com Typescript e Node</title>

    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link href=\\"/tailwind.css\\" rel=\\"stylesheet\\">
  </head>

  <body>
    <header class=\\"flex flex-row justify-between border-b p-3\\">
  Template MVC com Typescript e Node
  <div class=\\"relative\\">
    <input type=\\"checkbox\\" id=\\"language-selector\\" class=\\"hidden absolute\\">
    <label for=\\"language-selector\\" class=\\"flex items-center space-x-1 cursor-pointer\\">
      <span>pt-br</span>
      <svg class=\\"h-4 w-4\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\" />
      </svg>
    </label>

    <div id=\\"languages-menu\\" class=\\"absolute mt-1 right-1 top-full min-w-max shadow rounded opacity-0 border border-gray-400 transition delay-75 ease-in-out z-10 bg-white\\">
      <ul class=\\"block text-right\\">
        
          <li><a href=\\"?lang=en\\" class=\\"block px-3 py-2\\">en</a></li>
        
          <li><a href=\\"?lang=pt-br\\" class=\\"block px-3 py-2\\">pt-br</a></li>
        
      </ul>
    </div>
  </div>
</header>


    <div class=\\"p-3\\">
     
Ol√° Mundo! Rodando em test

    </div>
  </body>
</html>
"
`;
